### DISCUSSION TITLE
Initialize data inside SSR NextJs

### DISCUSSION THREAD

**Discussion by dextersiah** [pmndrs/zustand#1458] [General]
Is it possible to initialize the store within NextJs getServerSideProps function? 

I'm trying to execute the store within the function however I'm getting an error

`Warning: Invalid hook call. Hooks can only be called inside of the body of a function component.`

Based on the error it seem like zustand under the hood is using hooks which is not available on the server.

Ideally within my initEmployerStore I would be calling an API to fetch my user data.

```
export const getServerSideProps = async () => {

  const store = employerStore();

  store.initEmployerStore();

  return {
    props: {
      store
    }
  }
}
```

**Comment by dai-shi** [comment]
While I wonder if updating store in getServerSideProps is a bad practice, you should be able to do `const store = employerStore.getState();` to avoid the hook error.

**Reply by dextersiah** [reply]
I'm attempting to create a user authentication HOC which will initiate the store if the user is logged in but redirects the user if not logged in. I would assume that ideally it is best to initialize the store on the server side?

Do u have any suggestions on this?

**Reply by dai-shi** [reply]
My understanding is that we should use context in SSR to avoid conflicts between renders.
https://github.com/pmndrs/zustand/blob/bf9d0922fe9292c2399b9594eafe9304016b52d9/docs/guides/initialize-state-with-props.md might help.

**Reply by dextersiah** [reply]
Will take a look into that thanks for the help @dai-shi
